Add new "includes" verification check for multi-value verification

duke8253 opened the pull request:
<!-- The following line must be included in your pull request -->
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

bneradt reviewed (changes requested):
So this removes order verification for duplicate fields? I don't think we can remove order verification as a feature from Proxy Verifier because, while order of different field names is not significant, order of duplicate field names with various values (such as `Set-Cookie` in the example) is significant:

https://www.rfc-editor.org/rfc/rfc2616#section-4.2

>    Multiple message-header fields with the same field-name MAY be
>   present in a message if and only if the entire field-value for that
>   header field is defined as a comma-separated list [i.e., #(values)].
>   It MUST be possible to combine the multiple header fields into one
>   "field-name: field-value" pair, without changing the semantics of the
>   message, by appending each subsequent field-value to the first, each
>   separated by a comma. The order in which header fields with the same
>   field-name are received is therefore significant to the
>   interpretation of the combined field value, and thus a proxy MUST NOT
>   change the order of these field values when a message is forwarded.

It has been a few years, but I think when Alan asked me to work on this, we agreed that order verification was significant.

Having said that, maybe we can add a flag to ignore the order if the user desires.

duke8253 commented:
Sorry I should have been more clear on this, this change only applies to the `contains` check, I've updated the README and added a new test to reflect this.

bneradt commented on test/autests/gold_tests/field_verification/replay_files/multi_value_contains.yaml:
```diff
@@ -0,0 +1,50 @@
+meta:
```
We'll need the Apache license header.

bneradt reviewed (changes requested):
Thanks for clarifying. Looks good! Just one comment about needing a license.

bneradt reviewed (approved):
Looks like other yaml files are missing the license. We can add them as a separate PR.

bneradt commented on README.md:
```diff
@@ -1041,6 +1041,7 @@ equal      |  The presence of a field with the specified field name and a value
 contains   |  The presence of a field with the specified name with a value *containing* the specified value.
 prefix     |  The presence of a field with the specified name with a value *prefixed* with the specified value.
 suffix     |  The presence of a field with the specified name with a value *suffixed* with the specified value.
+includes   |  The presence of a field with the specified name with a value *including* the specified value. This only applies to multi-value field verification.
```
Maybe something like:

```
Helpful for multi-value fields. Specifies that the set of value strings exist in the header
field values for a particular header name in any order. Note that each value is matched
against the set like ``contains``, so each value is a substring match.
```

bneradt reviewed (changes requested):
> any rder
